sysdig/resource_sysdig_secure_organization.go

```diff
@@ -0,0 +1,154 @@
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(timeout),
+			Update: schema.DefaultTimeout(timeout),
+			Read:   schema.DefaultTimeout(timeout),
+			Delete: schema.DefaultTimeout(timeout),
+		},
+		Schema: map[string]*schema.Schema{
+			"cloud_provider_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cloud_provider_type": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{proto.Provider_PROVIDER_AWS.String(), proto.Provider_PROVIDER_GCP.String(), proto.Provider_PROVIDER_AZURE.String()}, false),
+			},
+			"customer_id": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"organization_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func getSecureOrganizationClient(c SysdigClients) (v2.OrganizationSecureInterface, error) {
+	return c.sysdigSecureClientV2()
+}
+
+func resourceSysdigSecureOrganizationCreate(ctx context.Context, data *schema.ResourceData, i interface{}) diag.Diagnostics {
+	client, err := getSecureOrganizationClient(i.(SysdigClients))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	org := secureOrganizationFromResourceData(data)
+
+	orgCreated, err := client.CreateOrganizationSecure(ctx, &org)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	data.SetId(orgCreated.Id)
+
+	return nil
+}
+
+func resourceSysdigSecureOrganizationDelete(ctx context.Context, data *schema.ResourceData, i interface{}) diag.Diagnostics {
+	client, err := getSecureOrganizationClient(i.(SysdigClients))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = client.DeleteOrganizationSecure(ctx, data.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func resourceSysdigSecureOrganizationRead(ctx context.Context, data *schema.ResourceData, i interface{}) diag.Diagnostics {
+	client, err := getSecureOrganizationClient(i.(SysdigClients))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	org, err := client.GetOrganizationSecure(ctx, data.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = secureOrganizationToResourceData(data, org)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+func resourceSysdigSecureOrganizationUpdate(ctx context.Context, data *schema.ResourceData, i interface{}) diag.Diagnostics {
+	client, err := getSecureOrganizationClient(i.(SysdigClients))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	org := secureOrganizationFromResourceData(data)
+
+	_, err = client.UpdateOrganizationSecure(ctx, data.Id(), &org)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+/* TODO: Make sure these are right inputs for the api */
+func secureOrganizationFromResourceData(data *schema.ResourceData) v2.OrganizationSecure {
+	return v2.OrganizationSecure{
+		Id:         data.Get("organization_id").(string),
+		ProviderId: data.Get("cloud_provider_id").(string),
```

Comment:
@mdkulkarni15 Do we know what exactly is the `providerId` here? For single account case, I've seen `providerId` set as the `accountId` when we run cloudauth API `/api/cloudauth/v1/accounts` today.
cc: @raghavanandan

Reply:
Right now the api is very aws specific, we need to come up with what fields are required for all the cloud providers. We can discuss this holistically. Id is subscription id/ tenantid/ folderid etc.. providerId is account id in accounts api

Reply:
Thanks Mandar.  2 questions -
- In that case, this API will work for GCP too then right? Because GCP is the first support we are adding this for.
- Isn't `id` a computed field for `/organizations` API as well (similar to `/accounts` api)

Reply:
let us wait till Haresh makes the organizations api more generic, as i said it is very aws specific. Once we finalize the api this change should be very trivial

Reply:
Fixed in the latest push.